engine/src/test/java/io/zeebe/engine/util/EngineRule.java

```diff
@@ -315,8 +324,13 @@ private void onNewEventCommitted() {
             final DeploymentRecord deploymentRecord = new DeploymentRecord();
             deploymentRecord.wrap(buffer);
 
-            environmentRule.writeCommandOnPartition(
-                partitionId, key, DeploymentIntent.CREATE, deploymentRecord);
+            // we run in processor actor, we are not allowed to wait on futures
+            // which means we cant get new writer in sync way
+            new Thread(
```

Comment:
I also get why we do this, but it feels a little hacky. The only alternative I can think of though is having async write APIs for tests as well, so you could them in cases like this (assuming here we would have access to the actor we're running in):

```java
actor.runOnCompletion(environmentRule.writeCommandOnPartitionAsync(partitionId, key, DeploymentIntent.CREATE, deploymentRecord), (result, error) -> CompletableActorFuture.completed(null));
```

But yeah not sure how much work it would be and how worth it it is :s

Reply:
on the next use case we create a separate api for that :crossed_fingers: